packages/circuits/src/lib/BytePacking.circom

```diff
@@ -0,0 +1,58 @@
+pragma circom 2.0.0;
+
+include "../../node_modules/circomlib/circuits/comparators.circom";
+include "../../node_modules/circomlib/circuits/bitify.circom";
+
+/// @dev when dealing with Num2Bits in circom, LSB is stored in index 0, MSB in last index
+/// @dev therefore when converting from bits to numbers (Bits2Num) and vice versa (Num2Bits)
+/// @dev we need to first reverse the order of the cards
+template UnpackCards(n) {
+
+    signal input packedCards[n];
+    signal output unpackedCards[n*31];
+    
+    // unpack cards into bytes
+    component lt[n*31];
+    for (var i = 0; i < n; i++) {
+        var sum = 0;
+        var mult = 1;
+        for (var j = 0; j < 31; j++) {
+            unpackedCards[(i*31) + (30-j)] <-- (packedCards[i] >> (j*8)) & 255;
```

Comment:
So we're assuming we're packing the cards in big-endian order? Such that if there's say 41 cards in a deck, the second felt will have its 10 high order bytes sets and the other ones zeroed.

Is there a reason? Not a big deal, but packing low-endian feels a little bit more intuitive to me.

Reply:
Yeah little endian makes sense, have made the changes!